api/users/serializers.py

```diff
@@ -104,6 +104,17 @@ def update(self, instance, validated_data):
             instance.save()
         except ValidationValueError as e:
             raise InvalidModelValueError(detail=e.message)
+        # ValidationError s raised by instance.save() are likely to have e.args[0] be
+        # dict of the form {'field validated': [<Exception>]}
+        except ValidationError as e:
+            if isinstance(e.args[0], dict):
+                value = e.args[0].popitem()[1]
+                if len(value) == 1:
+                    if isinstance(value[0], Exception):
```

Comment:
What if `value[0]` isn't an exception? Are there other cases to handle, i.e. a string? Should there be an `else`?